python/ray/dataframe/dataframe.py

```diff
@@ -3815,3 +4026,117 @@ def iloc(self):
         raise NotImplementedError(
             "To contribute to Pandas on Ray, please visit "
             "github.com/ray-project/ray.")
+
+    def _copartition(self, other, new_index):
```

Comment:
This method repartitions the two DFs so that they have the same partitioning?

Reply:
Yes, based on the index. I will add more detailed notes here.